src/source/tcp.rs

```diff
@@ -43,14 +42,21 @@ pub trait TCPStreamHandler: 'static + Default + Clone + Sync + Send {
         Default::default()
     }
 
+    /// Invokes handle_stream, toggling readiness if the function returns control.
+    fn run(&mut self, chan: util::Channel, poll: &mio::Poll, stream: mio::net::TcpStream) -> () {
```

Comment:
This function needs to be cleaned up.  We can resume just calling handle_stream directly.

Reply:
Removed.